internal/pkg/skuba/addons/addons.go

```diff
@@ -119,14 +121,27 @@ func (renderContext renderContext) ManifestVersion() string {
 
 // registerAddon incorporates one addon information to the Addons map that keeps track of the
 // addons which will get deployed
-func registerAddon(addon kubernetes.Addon, addonTemplater addonTemplater, preflightAddonTemplater preflightAddonTemplater, callbacks addonCallbacks, addonPriority addonPriority, getImageCallbacks []getImageCallback) {
-	Addons[addon] = Addon{
-		addon:              addon,
-		templater:          addonTemplater,
-		preflightTemplater: preflightAddonTemplater,
-		callbacks:          callbacks,
-		addonPriority:      addonPriority,
-		getImageCallbacks:  getImageCallbacks,
+func registerAddon(addon kubernetes.Addon, isCniAddon bool, addonTemplater addonTemplater, preflightAddonTemplater preflightAddonTemplater, callbacks addonCallbacks, addonPriority addonPriority, getImageCallbacks []getImageCallback) {
+	if isCniAddon {
```

Comment:
It scapes to me totally why you need two maps of addons with the same structure. Maybe will be enough and you can filter by addon type.

Reply:
My thinking was that maintaining 2 maps would make it easier to perform logic specific to CNI plugins. On second thought it really is a bad idea. What is really needed is logic that filters addons, so I'll continue maintaining a single map and develop that addon filtering logic.